src/sentry/api/endpoints/project_details.py

```diff
@@ -559,7 +560,14 @@ def put(self, request, project):
                 ]
         if result.get("symbolSources") is not None:
             if project.update_option("sentry:symbol_sources", result["symbolSources"]):
-                changed_proj_settings["sentry:symbol_sources"] = result["symbolSources"] or None
+                # Redact secrets so they don't get logged directly to the Audit Log
+                sources_json = result["symbolSources"] or None
+                try:
+                    sources = parse_sources(sources_json)
+                except Exception:
+                    sources = []
+                redacted_sources = redact_source_secrets(sources)
+                changed_proj_settings["sentry:symbol_sources"] = redacted_sources
```

Comment:
but isn’t this the main place where project details are saved?

Reply:
No, that's done on line 562: `project.update_option()`.  This variable is only used for the audit log.

Reply:
yep, to expand on what floris has said project option updating in this method has the following pattern:
- if the request includes an option that needs to be updated
- try to update the option
- if the update succeeded (returns true) then record the new value in `changed_proj_settings`